lib/chef/provider/package/deb.rb

```diff
@@ -111,7 +111,7 @@ def reconfig_package(name)
             # Runs command via shell_out with magic environment to disable
             # interactive prompts.
             def run_noninteractive(*command)
-              shell_out!(*command, env: { "DEBIAN_FRONTEND" => "noninteractive" })
+              shell_out!(*command, env: { "DEBIAN_FRONTEND" => "noninteractive" }.merge(new_resource.environment))
```

Comment:
This is the deb provider, but you added this to the apt resource - they are not the same. You need to do the deb resource and deb provider and/or apt resource and apt provider.

Reply:
i get what you mean, i'm writing a fix

Reply:
Yes, you're correct, deb is a helper module, so this applies to both apt_package and dpkg_package.